x-pack/plugins/global_search_providers/public/providers/get_app_results.ts

```diff
@@ -43,13 +75,17 @@ export const scoreApp = (term: string, { title }: PublicAppInfo): number => {
   return 0;
 };
 
-export const appToResult = (app: PublicAppInfo, score: number): GlobalSearchProviderResult => {
+export const appToResult = (
+  app: PublicAppInfo,
+  score: number,
+  subLink?: PublicAppSubLinkInfo
+): GlobalSearchProviderResult => {
   return {
-    id: app.id,
-    title: app.title,
+    id: subLink ? `${app.id}-${subLink.id}` : app.id,
+    title: subLink?.title ?? app.title,
     type: 'application',
```

Comment:
Do we want these sublinks to be of the `application` type, or do we want a new type, such as `feature` or something? Keeping `application` is easier to integrate with the searchbar's per-type logic, but I'm wondering if we may want to dissociate these results from the base/current `application` ones?

Reply:
To me, it felt like it made sense to have the scoring for these done in a single place so that the more specific sublinks are ranked higher than the general applications when possible. Happy to experiment with a separate provider if we feel that is better.